src/hotspot/share/runtime/thread.hpp

```diff
@@ -644,13 +644,13 @@ class Thread: public ThreadShadow {
   }
 #endif // __APPLE__ && AARCH64
 
- // support ASGCT
+ // support AGCT
  private:
-  bool _in_asgct;
+  bool _in_agct;
```

Comment:
This should actually be in JavaThread as AGCT only operates on JavaThreads.

Reply:
I will have to do check/cast in `CodeCache::find_blob()` as that may get called from any thread, not just Java threads.
I would assume that having this flag defined at Thread level is a lesser of the evils - or am I wrong here?

Reply:
We have been actively moving JavaThread fields out of Thread.

Reply:
Ok. Moved to `JavaThread`.
There is a small cost to pay in `CodeCache::find_blob()` where we need to check if the thread is a `JavaThread` because that function can be called also from non-java threads, AFAIK.